apps/rush-lib/src/cli/test/RushCommandLineParser.test.ts

```diff
@@ -279,4 +282,63 @@ describe('RushCommandLineParser', () => {
       });
     });
   });
-});
\ No newline at end of file
+
+  describe(`in repo with tests for add`, () => {
+    describe(`'add' action`, () => {
+      it(`adds a dependency to just one sub-repo`, () => {
+        const startPath: string = resolve(__dirname, 'addRepo');
+        const aPath: string = resolve(__dirname, 'addRepo/a');
+        FileSystem.deleteFile(resolve(__dirname, `addRepo/a/package-deps.json`));
+
+        // Create a Rush CLI instance. This instance is heavy-weight and relies on setting process.exit
+        // to exit and clear the Rush file lock. So running multiple `it` or `describe` test blocks over the same test
+        // repo will fail due to contention over the same lock which is kept until the test runner process
+        // ends.
+        const parser: RushCommandLineParser = new RushCommandLineParser({ cwd: startPath });
+
+        // Switching to the "a" package of addRepo
+        process.chdir(aPath);
+
+        // Mock the command
+        process.argv = ['pretend-this-is-node.exe', 'pretend-this-is-rush', 'add', '-p', 'assert'];
+
+        expect.assertions(2);
+        return expect(parser.execute()).resolves.toEqual(true)
+          .then(() => {
+            const packageJSON = FileSystem.readFile(resolve(__dirname, `addRepo/a/package.json`));
+            expect(packageJSON).toEqual(expect.stringMatching('"assert"'));
+          });
+      });
+    });
+
+    describe(`'add' action with --all`, () => {
+      it(`adds a dependency to just one sub-repo`, () => {
```

Comment:
This needs to be named differently to capture the expected result of this test.

Reply:
Hi! How would you name it? I'm out of ideas right now.

Reply:
I followed your recommendations (sent through another chat). Let me know if you have any other feedback 💙